### DISCUSSION TITLE
File Editor su HM core. Problema accesso file YAML

### DISCUSSION THREAD

**Discussion by Asky82** [indomus/forum#2847] [HUB: Home Assistant]
Ciao a tutti,

Ho installato HA core su un mini PC consapevole che questa versione non ha by default l'accesso agli adds on ma questi vanno installati manualmente.

Al momento ero interessato ad avere la possibilità di editare il file di configurzione YAML via HA con File Editor ma non ho trovato nessuna guida su come installarlo manualmente su core. Posso chiedere se esite qualcolsa che possa aiutarmi?

O in generale a capire come installare manualmente addon ( anche trovarli) su HM core? 

Grazie 1000

**Comment by Scialla** [comment]
Usa config Editor su hacs

**Reply by Asky82** [reply]
Grazie! 
Sto provando ad implementare config editor ma sto riscontrando un altro problema. 

In un punto mi chiede di chiamare il servizio nel configuration yaml ma purtroppo quando modofico il file e tento di salvarlo mi appare l'errore "access denied".  :(

Sto guardano in rete per una soluzione ma niente per il momento...

**Reply by Asky82** [reply]
Anche questa è risolta. Ho aggiunto l'utente al gruppo "root". Spero di non aver fatto una cavolata :)

**Reply by snakuzzo** [reply]
Infatti hai fatto una cavolata. Questa cosa dei permessi e proprietà dei file la prendete un po' tutti alla leggera.
Magari cambi i permessi e owner dei file, ma non è che metti il tuo utente nel gruppo root

**Reply by Asky82** [reply]
Ok ok  :) passo indietro. Ho rimosso user da root e modificato permessi al file da only write to read and write per tutti. 

Grazie per la dritta. Sto imparando da zero linux includendo la configurazione di HA core. E due settimane che ci lavoro per configurarlo come voglio ma è un pò dura. In ogni modo giorno dopo giorno si fanno passi in avanti..

**Reply by snakuzzo** [reply]
Già va meglio :) ...ma in realtà ti conviene fare diversamente...ovvero cambiare ricorsivamente l'owner dei file

```
sudo chown -R <tuoutente>:<tuogruppo> <directory>
```

P.S. Considera che ogni nuovo file creato direttamente dal sistema (esempio i file sotto `.storage`, che comunque tu non andrai mai a modificare) la prima volta verranno creati con owner `root:root`

**Comment by Asky82** [comment]
Perdonami ma in directory cosa devo inserire? il percorso del file + file name?

**Reply by snakuzzo** [reply]
Per favore non creare nuovi thread nella stessa discussione.

`<directory>` sarebbe la directory nella quale sono presenti i file a cui vuoi cambiare l'ownership

**Reply by Asky82** [reply]
Assolutamente, perdonami. 

E grazie ancora per le dritte ;)
